### ISSUE TITLE
Validate range-words in Form component (UI Component)

### ISSUE THREAD

**Issue by robinhuy** [magento/magento2#6113]
I cannot configure a form field with validation `range-words`, it different than `validate-number-range`

### Preconditions

1. Magento 2

### Steps to reproduce

1.  Edit {Magento root directory}/app/code/Magento/Catalog/view/adminhtml/ui_component/new_category_form.xml:
In node with '<field name="name"' replace                      

`<rule name="required-entry" xsi:type="boolean">true</item>`

 to

`<rule name="range-words" xsi:type="array">`
`        <item name="0" xsi:type="number">1</item>`
`        <item name="1" xsi:type="number">3</item>`
`</rule>`

2. Flush cache.
3. In Admin, open any product and click "New Category"
4. Enter any value in "Category Name" field, set any "Parent Category" and press "Create Category"

### Expected result

Category Name is validated and Category is created (or correct error message is displayed, if validation fails)

### Actual result

Nothing happens explicitly. Error message appers in browser console.

When I test above code, it fired a bug in Chrome Console:
`rules.js:78 Uncaught TypeError: Cannot read property 'length' of null`

In rules.js

```
"range-words": [
    function(value, params) {
    return utils.stripHtml(value).match(/\b\w+\b/g).length >= params[0] &&
        value.match(/bw+b/g).length < params[1];
    },
    $.mage.__('Please enter between {0} and {1} words.')
]
```

Why the regex pattern check with params[0] and params[1] is different?

This is my fault or Magento bug and how to use `range-words` validation ?

**Comment by veloraven** [comment]
@robinhuy thank you for your report.
Please, provide the used Magento version.

**Comment by robinhuy** [comment]
@veloraven: I'm using Magento version 2.1 clone from github

**Comment by picamator** [comment]
@robinhuy  thank you for giving details about version.

You are correctly using validator. The regex patterns should be the same.
We've created internal ticket MAGETWO-57147 to fix it.

**Comment by magento-engcom-team** [comment]
@robinhuy, thank you for your report.
We've created internal ticket(s) MAGETWO-84403 to track progress on the issue.

**Comment by AndrewMun** [comment]
#mageconf

**Comment by okorshenko** [comment]
Hi @robinhuy. Thank you for your report. 
The issue has been fixed in magento/magento2#12739 by @zamoroka in 2.2-develop branch
Related commits:
 - 465e14bdee949bf8c92ed1640119930e0cce4091
 - 8bc807fb097ff2ef8f45ff756d69e27f563cf388
 - e9dcb03d2430e0b4289502c3550c0e24fd945e3b

The fix will be available with the upcoming patch release.

**Comment by okorshenko** [comment]
Hi @robinhuy. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1298 by @magento-engcom-team in 2.3-develop branch
Related commit(s):
 - 54301a8ea63134b961d6b7b9891c31bc2798f8dc
 - 402b719168fc8b17ebc099c12b85f629cf90303b
 - 3837f2780b32112b7f70a9d3074158ee2336b1b7
 - 031e23e93ffaa258a39dfed5c5edb0e873ab0aa8
 - aa267f462f7eb964226a9c89d20c04b718406337
 - 934818dbee0bd36ec5eae7b3b7230c4d8a70b481
 - 6bed7d475bff5e98e27b8fdcaa998438bc31ed05
 - 72b2794043afd0012f9a8d5f6fc04df07acad3af
 - 4e64c415e26845826b527c17008eda00d3808591
 - d3feb280ed621c5ffe2dc3a2b7ac30abb25d9c22
 - 7aeccea46d2336a1322811a9d60d52432108c8e3
 - b857c022ae1cdb7975cf5c3d79cd679843c72a3b
 - a75bf2391dfc6e7c8a1283490f5de7551de42bbe
 - ab2e5798e529633f770c3970a0df999c11764e06
 - d904690443d43322c5dd2c58d075944eae867316
 - a821c7a3c17eb06983f421c47a7cc0fe99983a38
 - 68ac4c0bd6cae6b5b62e683c23f49293ee41d797
 - 93fe4d14faba635cc7c668b2d931e7595567cda9
 - f43fd4b98c55e1f498d060aeb245e4c969bdcb13
 - 2c52a8fd15cff7af7245573e2f26be6d601acb08

The fix will be available with the upcoming 2.3.0 release.
